src/main/java/se/sundsvall/seabloader/service/mapper/InvoiceMapper.java

```diff
@@ -57,4 +74,30 @@ public static InExchangeInvoiceStatusType toInExchangeInvoice(final String xml)
 		final var unmarshaller = JAXBContext.newInstance(InExchangeInvoiceStatusType.class).createUnmarshaller();
 		return (InExchangeInvoiceStatusType) JAXBIntrospector.getValue(unmarshaller.unmarshal(xmlSource));
 	}
+
+	private static InvoicePdf toInvoicePdf(final InExchangeInvoiceStatusType inExchangeInvoiceStatusType) {
```

Comment:
The InvoicePdfMerger does this already, no need to have all this logic here. Why not just autowire in the InvoicePdfMerger (it is written as a Spring component), have the initial null check on both **originalInvoice** and **attachments** and then call mergePdfs with inExchangeInvoiceStatusType as argument, and finally return a InvoicePdf with name from inExchangeInvoice.getOriginalinvoice().getName() and the response from the merger encoded as Base64 string?